lib/active_merchant/billing/gateways/paybox_direct.rb

```diff
@@ -63,11 +63,30 @@ def initialize(options = {})
         requires!(options, :login, :password)
         super
       end
-
+      
+      def add_3dsecure(post, three_d_secure)
+        if three_d_secure.eci == '02' || three_d_secure.eci == '05'
+            post[:3DSTATUS]  = 'Y'
+            post[:3DENROLLED]  = 'Y'
+            post[:3DSIGNVAL] = 'Y'
+            post[:3DERROR] = '0'
+        else 
+            post[:3DSTATUS]  = 'N'
+            post[:3DENROLLED]  = 'N'
+            post[:3DSIGNVAL] = 'N'
+            post[:3DERROR] = '10000'
+        end
+        post[:3DECI]  = three_d_secure.eci
+        post[:3DXID] = three_d_secure.xid
+        post[:3DCAVV] = three_d_secure.cavv
+        post[:3DCAVVALGO] = '0000000001'
+      end
+      
       def authorize(money, creditcard, options = {})
         post = {}
         add_invoice(post, options)
         add_creditcard(post, creditcard)
+        add_3dsecure(post, three_d_secure)
```

Comment:
`three_d_secure` does not exist in this context. If you want to know how to get the 3DS data, you can have a look here: https://github.com/activemerchant/active_merchant/wiki/Standardized-3DS-Fields